src/idl/tests/annotation.c

```diff
@@ -455,93 +455,152 @@ CU_Test(idl_annotation, redefinition)
+    {"@autoid(HASH) struct s { char c; @hashid(\"c\") char d; };",                  IDL_RETCODE_SEMANTIC_ERROR, {0},              {0},            {0}}                                      //clashing hashid fields
+  };
 
-#if 0
-CU _ Test(idl_annotation, autoid_struct)
+  for (size_t i = 0; i < sizeof(tests)/sizeof(id_test_t); i++)
+    test_id(tests[i]);
+}
+
+CU_Test(idl_annotation, autoid_inheritance)
 {
-  idl_retcode_t ret;
-  idl_tree_t *tree = NULL;
-  idl_struct_t *s1;
-  const char str[] = "@autoid struct s { char c; };";
+  static const id_test_t tests[] = {
+    {"struct base { char c; char d; };\n"
+     "struct derived : base { char e; char f; };",                IDL_RETCODE_OK,             {IDL_SEQUENTIAL, IDL_SEQUENTIAL},                 {0},  {0, 1, 2, 3}},                                          //implicit sequential autoid, single inheritance
+    {"struct base { char c; char d; };\n"
+     "struct derived : base { char e; char f; };\n"
+     "struct morederived : derived { char g; char h; };",         IDL_RETCODE_OK,             {IDL_SEQUENTIAL, IDL_SEQUENTIAL, IDL_SEQUENTIAL}, {0},  {0, 1, 2, 3, 4, 5}},                                    //implicit sequential autoid, double inheritance
+    {"@autoid(HASH) struct base { char c; char d; };\n"
+     "struct derived : base { char e; char f; };",                IDL_RETCODE_OK,             {IDL_HASH, IDL_SEQUENTIAL},                       {0},  {0x00088a4au, 0x01e07782u, 0x01e07783u, 0x01e07784u}},  //explicit hash autoid, derived class counting sequentially
+    {"@autoid(SEQUENTIAL) struct base { char c; char d; };\n"
+     "@autoid(HASH) struct derived : base { char e; char f; };",  IDL_RETCODE_OK,             {IDL_SEQUENTIAL, IDL_HASH},                       {0},  {0, 1, 0x071767e1u, 0x0d4ca18fu}},                      //explicit sequential autoid, derived class hashing
+    {"struct base { char c; char d; };\n"
+     "struct derived : base { @id(1) char e; char f; };",         IDL_RETCODE_SEMANTIC_ERROR, {0},                                              {0},  {0}},                                                   //field in derived class clashes
+    {"@autoid(HASH) struct base { char c; char d; };\n"
+     "struct derived : base { @hashid(\"c\") char e; char f; };", IDL_RETCODE_SEMANTIC_ERROR, {0},                                              {0},  {0}}                                                    //field in derived class clashes
+  };
 
-  ret = idl_parse_string(str, IDL_FLAG_ANNOTATIONS, &tree);
-  CU_ASSERT_EQUAL_FATAL(ret, IDL_RETCODE_OK);
-  CU_ASSERT_PTR_NOT_NULL(tree);
-  s1 = (idl_struct_t *)tree->root;
-  CU_ASSERT_PTR_NOT_NULL_FATAL(s1);
-  CU_ASSERT_FATAL(idl_is_struct(s1));
-  CU_ASSERT_EQUAL(s1->autoid, IDL_AUTOID_SEQUENTIAL);
-  idl_delete_tree(tree);
+  for (size_t i = 0; i < sizeof(tests)/sizeof(id_test_t); i++)
+    test_id(tests[i]);
 }
 
-// x. autoid twice
-// x. autoid (HASH)
-// x. autoid (SEQUENTIAL)
+CU_Test(idl_annotation, autoid_module)
```

Comment:
These tests aren't a good fit for the `test_id` function IMHO. I'd like to see some that try to break the assignment of ids. So different branches etc.

Reply:
Still like to see tests with some additional modules.

Reply:
I will create more tests for autoid on modules